samples/drivers/adc/src/main.c

```diff
@@ -22,149 +29,58 @@ static const struct adc_dt_spec adc_channels[] = {
 			     DT_SPEC_AND_COMMA)
 };
 
-#define LABEL_AND_COMMA(node_id, prop, idx) \
-	DT_LABEL(DT_IO_CHANNELS_CTLR_BY_IDX(node_id, idx)),
-
-/* Labels of ADC controllers referenced by the above io-channels. */
-static const char *const adc_labels[] = {
-	DT_FOREACH_PROP_ELEM(DT_PATH(zephyr_user), io_channels,
-			     LABEL_AND_COMMA)
-};
```

Comment:
👍 Thanks a lot for simplifying this.

I don't know how I could miss this that I could just use `dev->name` instead of complicating things this way... 🤦